capstone/src/main/java/com/google/step/servlets/RestaurantsServlet.java

```diff
@@ -14,55 +14,66 @@
 
 package com.google.step.servlets;
 
-import com.google.appengine.api.datastore.GeoPt;
-import com.google.gson.Gson;
-import com.google.gson.JsonObject;
-import com.google.step.clients.RestaurantClient;
-import com.google.step.data.Restaurant;
+import com.google.api.client.json.Json;
+import com.google.step.data.RestaurantHeader;
+import com.google.step.search.ElasticsearchClient;
+import com.google.step.search.RestaurantHeaderSearchClient;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import org.json.JSONArray;
-import org.json.JSONObject;
 
-/** Servlet responsible for getting restaurants from Datastore. */
+/** Servlet responsible for getting restaurants from the search index */
 @WebServlet("/restaurants")
 public class RestaurantsServlet extends HttpServlet {
-  private RestaurantClient restaurantClient = new RestaurantClient();
+  private final RestaurantHeaderSearchClient searchClient;
+
+  public RestaurantsServlet() {
+    this(new ElasticsearchClient("localhost", (short)9200));
+  }
+
+  RestaurantsServlet(RestaurantHeaderSearchClient searchClient) {
+    this.searchClient = searchClient;
+  }
 
   /**
-   * Returns a list of RestaurantHeaders, with a snapshot of restaurant details.
-   * @param response A list of restaurant details, in the following json format:
-    {
-      "restaurantHeaders" : [
-        {
-          "name": <String>,
-          "cuisine": <List<String>>,
-          "message": <String, details status of restaurant based on score>,
-        },
-        ...
-      ]
-    }
-   * @param request Specifies relevant search params. //TODO: update exact search param format
-   * @throws //TODO: add any error codes thrown once search is implemented.
+   * Request headers from the search server for restaurants without any particular query
+   * @param response  a servlet response with a JSON object holding restaurant details in the
+   *                 response body. The restaurant details can be found in the "restaurants" field
+   * @param request simple get request. Takes no parameters
+   * @throws IOException thrown if writing the response fails
    */
-  // TODO: Change this to be connected to search functionality.
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<Restaurant> restaurants = restaurantClient.getRestaurantsNoFilter();
+    List<RestaurantHeader> restaurantHeaders = searchClient.searchRestaurants("");
+    JSONObject responseJson = new JSONObject()
+        .put("restaurants", new JSONArray(restaurantHeaders));
+
+    response.setContentType(Json.MEDIA_TYPE);
```

Comment:
Will this response JSON be parsed in the frontend the same way as the old response type or will we have to change that code slightly?

Reply:
We'll have to change the code a bit because in this case, because there is no information about whether a restaurant is struggling or not. Once we have scoring stuff, we'll probably have to put in place some cutoffs or something, but I'm not really sure.